src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -291,17 +349,44 @@ public void filterForExam(StudentExam studentExam, Exercise exercise, List<Stude
     }
 
     /**
-     * we also need the quiz questions and it is not possible to load them in a generic way with the entity graph used
+     * Helper method which attaches the result to its participation.
+     * For direct automatic feedback during the exam conduction for {@link ProgrammingExercise, we need to attach the results.
+     * We also attach the result if the results are already published for the exam. See {@link StudentExamResource#areResultsPublishedYet(StudentExam)}
+     * @param studentExam the given studentExam
+     * @param participation the given participation of the student
+     * @param latestSubmission the latest
+     * @param isAtLeastInstructor
+     */
+    private void setResultIfNecessary(StudentExam studentExam, StudentParticipation participation, boolean isAtLeastInstructor) {
+        // Only set the result during the exam (direct automatic feedback) or after publishing the results
+        boolean studentAllowedToSeeResult = (studentExam.getExam().isStarted() && !studentExam.isEnded()) || studentExam.areResultsPublishedYet();
+        Optional<Submission> latestSubmission = participation.findLatestSubmission();
+
+        if ((studentAllowedToSeeResult || isAtLeastInstructor) && latestSubmission.isPresent()) {
+            // Set the latest result into the participation as the client expects it there for programming exercises
+            Result result = latestSubmission.get().getResult();
+            if (result != null) {
+                participation.setResults(Set.of(result));
+            }
+        }
+    }
+
+    /**
+     * Helper method to load the quiz questions.
+     * It is not possible to load them in a generic way with the entity graph used
```

Comment:
This sentence sounds weird :)

Reply:
See 6d6270a751b9ee1fa362a832a54a72c747ce1f42